src/Console/Output/Progress/ProgressOutputType.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpCsFixer\Console\Output\Progress;
+
+/**
+ * @internal
+ */
+final class ProgressOutputType
```

Comment:
*major*

to allow register custom ProgressOutput implementation, we should get rid of this enum class.
context: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6848#discussion_r1146888035

I'm OK to merge this PR without this change implemented, to move on forward. But we should not keep it in PR that will allow for custom implementation

Reply:
Changed my mind, I will keep it here and let's continue in #6842. Since it's `@internal` it can be safely changed/removed later, if we decide like that.

But in short: this class only describes built-in output types, so there's no need for custom output classes to be registered here, they are going to be handled [differently](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6842#discussion_r1160323860)

Reply:
overall, I do believe that we should have similar mechanism to https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Console/Report/FixReport/ReporterInterface.php#L24 , rather than centralised enum-like class aiming to collect all possible types